File: upgrade-user.ts
const USER_ID = '3c2085b7-de19-456a-8055-ffb22dd9cbb2';

// Generate token
const payload = JSON.stringify({ sub: USER_ID });
const b64 = Buffer.from(payload).toString('base64');
const token = `a.${b64}.b`;

const BASE_URL = 'http://127.0.0.1:5000';

async function makeRequest(method: string, endpoint: string, body?: any) {
  const url = `${BASE_URL}${endpoint}`;
  const options: any = {
    method,
    headers: {
      'Authorization': `Bearer ${token}`,
      'Content-Type': 'application/json',
    },
  };
  
  if (body) {
    options.body = JSON.stringify(body);
  }

  const response = await fetch(url, options);
  const data = await response.json();
  return { status: response.status, data };
}

async function main() {
  console.log('📱 Upgrading user to family plan...');
  console.log(`Token: ${token}\n`);

  try {
    console.log('1️⃣ Calling upgrade endpoint...');
    const upgradeRes = await makeRequest('POST', '/api/user/upgrade-to-family', {});
    console.log('✅ Upgrade Response:', JSON.stringify(upgradeRes.data, null, 2));
    console.log();

    // Small delay before checking status
    await new Promise(r => setTimeout(r, 500));

    console.log('2️⃣ Checking premium status...');
    const statusRes = await makeRequest('GET', '/api/user/premium-status');
    console.log('✅ Premium Status:');
    console.log(JSON.stringify(statusRes.data, null, 2));
    
    if (statusRes.data.planType === 'family' && statusRes.data.isPremium) {
      console.log('\n🎉 SUCCESS! User is now on family plan!');
    } else {
      console.log('\n⚠️ User plan not set correctly');
    }
  } catch (err) {
    console.error('❌ Error:', err);
  }
}

main();

File: upgrade-user.py
import requests
import json
import base64

USER_ID = "3c2085b7-de19-456a-8055-ffb22dd9cbb2"

# Generate token
payload = json.dumps({"sub": USER_ID})
b64 = base64.b64encode(payload.encode()).decode()
token = f"a.{b64}.b"

headers = {
    "Authorization": f"Bearer {token}",
    "Content-Type": "application/json",
}

print("📱 Upgrading user to family plan...")
print(f"Token: {token}\n")

# Upgrade to family plan
try:
    response = requests.post(
        "http://127.0.0.1:5000/api/user/upgrade-to-family",
        headers=headers,
        json={},
        timeout=5
    )
    print(f"✅ Upgrade Response: {response.status_code}")
    print(f"Body: {response.text}\n")
except Exception as e:
    print(f"❌ Error: {e}\n")

# Check premium status
print("Checking premium status...")
try:
    response = requests.get(
        "http://127.0.0.1:5000/api/user/premium-status",
        headers=headers,
        timeout=5
    )
    data = response.json()
    print(f"✅ Status Response: {response.status_code}")
    print(f"Premium Status:")
    print(json.dumps(data, indent=2))
except Exception as e:
    print(f"❌ Error: {e}")
